zip-0315.rst

```diff
@@ -1,9 +1,629 @@
+
+Following this recommendation improves both collective privacy and the user's
+individual privacy, by maximizing the size of the note anonymity set over time.
+
+The remainder of this specification assumes a wallet that follows the above
+recommendation, except where explicitly noted.
+
+A wallet MAY allow users to disable auto-shielding.
+
+
+Trusted and untrusted TXOs
+--------------------------
+
+A wallet SHOULD treat received TXOs that are outputs of transactions created
+by the same wallet, as trusted TXOs. Wallets MAY enable users to mark specific
+external transactions as trusted, allowing their received TXOs also to be
+classified as trusted TXOs.
+
+A wallet SHOULD have a policy that is clearly communicated to the user for
+the number of confirmations needed to spend untrusted and trusted TXOs
+respectively. The following confirmation policy is RECOMMENDED:
+
+* 10 confirmations, for untrusted TXOs;
+* 3 confirmations, for trusted TXOs.
+
+Rationale for the given numbers of confirmations
+''''''''''''''''''''''''''''''''''''''''''''''''
+
+The rationale for choosing three confirmations for trusted TXOs is that
+empirically, reorgs are usually less than three blocks.
+
+The consequences of attempting to spend a trusted TXO may be less severe in the
+case of a rollback than the consequences of attempting to spend an untrusted TXO.
+The value received from a trusted TXO should always be recoverable, whereas
+recovering value received from an untrusted TXO may require the user to request
+that funds are re-sent.
+
+
+Categories of TXOs according to spendability
+--------------------------------------------
+
+A TXO is *spendable*, relative to a given block chain and wallet state,
+if and only if all of the following are true in that state:
+
+* the TXO is unspent;
+* the TXO is not committed to be spent in another transaction created
+  by this wallet; and
+* the wallet has the TXO's spending key (for whatever protocol the TXO uses).
+
+A TXO is *confirmed-spendable*, relative to a given block chain and
+wallet state, if and only if all of the following are true in that state:
+
+* the wallet is synchronized; and
+* the TXO is spendable; and
+* either auto-shielding is disabled or the TXO is shielded; and
+* the TXO is trusted and has at least the required confirmations for
+  trusted TXOs, or it is untrusted and has at least the required
+  confirmations for untrusted TXOs.
+
+A TXO is *unconfirmed-spendable*, relative to a given block chain and
+wallet state, if and only if the TXO is spendable but is not
+confirmed-spendable in that state.
+
+A TXO is *watch-only* if and only if the wallet has its full viewing key
+(or address in the case of a transparent TXO) but not its spending key.
+
+A wallet MUST NOT attempt to spend a TXO that is not confirmed-spendable.
+
+Note: the definition of a TXO includes outputs in mempool transactions
+that are unconflicted from the perspective of the wallet.
+
+
+Reporting of balances
```

Comment:
I suggest writing this differently to improve readability.. 

First, define different kind of balances the ZIP considers.

Then explain how wallets should report them. This lets wallet developers or analysts to quickly locate the concepts when needed.

Reply:
The intended meanings of the balances can't be defined precisely without the concept of trusted and untrusted TXOs. This is a Specification section: it should be using precise definitions.

In order to define balances before trusted/untrusted TXOs within the Specification section, we would first have to define balances loosely, then have a diversion to defining trusted and untrusted TXOs, then go back to defining balances strictly. That's just more confusing, and it carries the risk that implementors will try to apply their own interpretation of what the loose definitions mean, which is undesirable because part of the point of ZIP 315 is to promote consistency between wallets.

A loose definition of balances should, however, be given in the Requirements section. Currently "confirmed-spendable" is used before definition [on line 98](https://github.com/zcash/zips/pull/607/files#diff-81a3ab3f008443ec864f316692e2e6a15808c87d22f0032a930ce76019a02a05R98) in "Requirements &mdash; Prompt accessibility of funds", and that's a clarity bug.